File: src/main/java/cloud/DeleteFileServlet.java
package cloud;

import java.io.File;

import java.io.IOException;
import java.util.function.Consumer;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import com.google.gson.Gson;

import cloud.bean.CreateDir;
import cloud.bean.DeleteFile;
import cloud.bean.Response;

@WebServlet("/deletefile")
public class DeleteFileServlet extends HttpServlet {

	@Override
	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {

		DeleteFile params = JsonUtil.fromJsonStream(req.getInputStream(), DeleteFile.class);

		if(params.paths.get(0).equals(Cons.Path.USER_DIR_STUB)) {
			params.paths.remove(0);
		}

		String path = FileUtil.getWholePath(Cons.Path.DATA_DIR, TokenUtil.getUsername(req.getParameter("token")),
				FileUtil.getWholePath(params.paths));

		System.out.print("DeleteFileServlet: path->" + path + "\n");

		FileUtil.deleteFile(new File(path));

		resp.getWriter().write(JsonUtil.toJson(new Response<>(CodeMessage.OK.code, CodeMessage.OK.message, null)));

	}
}

File: src/main/kotlin/cloud/servlet/DeleteFileServlet.kt
package cloud.servlet

import cloud.model.net.CodeMessage
import cloud.model.net.DeleteFile
import cloud.model.net.Response
import cloud.config.Cons
import cloud.manager.logger
import cloud.util.*
import java.io.File
import java.io.IOException
import jakarta.servlet.ServletException
import jakarta.servlet.annotation.WebServlet
import jakarta.servlet.http.HttpServlet
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse

@WebServlet("/deletefile")
class DeleteFileServlet : HttpServlet() {
    @Throws(ServletException::class, IOException::class)
    override fun doPost(req: HttpServletRequest, resp: HttpServletResponse) {
        val location = CloudFileUtil.getWholePath(JsonUtil.fromJsonReader(req.reader, DeleteFile::class.java).paths, TokenUtil.getUsername(req.getParameter("token")))
        val path = FileUtil.getWholePath(Cons.Path.DATA_DIR, location)
        logger.info {
            "path:$path"
        }
        val dataFile = File(path)
        //先删除预览图
        if (dataFile.isDirectory) {
            FileUtil.deleteFile(PreviewFileUtil.getPreviewParentFile(dataFile, File(Cons.Path.DATA_DIR)))
        } else {
            PreviewFileUtil
                    .getPreviewFile(dataFile, File(Cons.Path.DATA_DIR))
                    ?.also {
                        logger.info {
                            "PreviewFile:${it.absolutePath}"
                        }
                        FileUtil.deleteFile(it)
                    }
        }
        //整个文件夹或文件删除
        FileUtil.deleteFile(dataFile)

        resp.writer.write(JsonUtil.toJson(Response<Any?>(CodeMessage.OK.code, CodeMessage.OK.message, null)))
    }
}